### DISCUSSION TITLE
Using SWR with HTTP long-polling

### DISCUSSION THREAD

**Discussion by cameronelliott** [vercel/swr#1856] [Q&A] [answered]
Hello, new to SWR (and React, really), anyway, it seems like a great way to do 'fetchy' stuff in React.

I have a server-side application which implements API updates via HTTP long-polling. 
https://en.wikipedia.org/wiki/Push_technology#Long_polling

Long polling is simply where the server doesn't respond to the HTTP request until new-data or an update is available.

**I'm wondering if there are examples of using SWR with long-polling????**

It seems I could use SWR with a very small  polling interval, ie `refreshInterval = 1`,
but since the library seems so comprehensive, I thought it makes sense to ask about
this use case specifically.

Thanks,
Cameron

**Comment by shuding** [comment, marked as answer]
Hello there! 

Unfortunately SWR doesn't have a built-in way to handle long-polling. Using `refreshInterval: 1` is a good work around indeed. 

I think it makes sense to think about a good way to support long-polling in SWR, perhaps with #1263 (but that would still require some code to setup, this is something we need to have in mind when designing this new API). Fundamentally long-polling is something more similar to subscription than fetch: single instance, always connected and deduplicates others.
